Lib/http/client.py

```diff
@@ -689,6 +691,8 @@ def readline(self, limit=-1):
         result = self.fp.readline(limit)
         if not result and limit:
             self._close_conn()
+            if self.length:
+                raise IncompleteRead(result, self.length)
         elif self.length is not None:
             self.length -= len(result)
```

Comment:
```suggestion
        elif self.length is not None:
            self.length -= len(result)
            if self.length and not result.endswith(b'\n'):
                raise IncompleteRead(result)
```

Reply:
Oh, you’re right on both counts. I forgot the _elif_ statement misses the empty result case, so the logic would need more work.

And I didn’t notice it is usual to call __close_conn_ when raising exceptions. Apparently this lets the HTTPConnection make another request. I think this should be consistent for all exceptions (e.g. if _fp.readline_ raises OSError), but that’s a separate issue.